n2y/converter.py

```diff
@@ -163,18 +164,48 @@ def __init__(self, block):
         for key, value in block.items():
             self.__dict__[key] = value
 
+    def handler(self, annotation, target):
+        handler = {
+            'bold': [Strong([target]), Strong(target)],
+            'italic': [Emph([target]), Emph(target)],
+            'strikethrough': [Strikeout([target]), Strikeout(target)]
+        }
+        return handler[annotation]
+
+    def strip(self, target):
+        prependages = deque()
+        appendages = deque()
+        result = []
+        blank_space = [Space(), SoftBreak()]
+        nested_annotations = False
+        while target[0] in blank_space:
+            prependages.appendleft(target.pop(0))
+        while target[-1] in blank_space:
+            appendages.appendleft(target.pop(-1))
+        for key in ['strikethrough', 'italic', 'bold']:
+            if self.__dict__[key]:
+                handler = self.handler(key, target)
+                target = handler[0] if nested_annotations else handler[1]
+                nested_annotations = True
+        if prependages:
+            for i in prependages:
+                result.append(i)
+        result.append(target)
+        if appendages:
+            for i in appendages:
+                result.append(i)
+        return result
+
     def apply_pandoc(self, target):
         result = target
+        annotations = [self.bold, self.italic, self.strikethrough]
```

Comment:
Yes but only the bold, italic, and strikethrough annotations can't affect leading and trailing spaces in gfm so I figure only change those in order to make our conversion as close to the original document as possible, only making changes when absolutely neccessary.

Reply:
I like the idea of making the minimal number of changes to the original document as possible. We may not even need to trim the space for strikethrough _if_ struck out spaces are visible in Notion. I think we definitely need to handle bold and italic because bolded spaces look just like unbolded spaces and italicized spaces look just like unitalicized spaces.

Reply:
leading and trailing spaces confuse markdown renderers. In the case of italic, a leading space will add a bullet instead of making the word italic.

This is how vscode renders:
![image](https://user-images.githubusercontent.com/4360601/161583829-7f99ee2c-59be-4791-86b4-fc33172d9f77.png)

Reply:
> We may not even need to trim the space for strikethrough _if_ struck out spaces are visible in Notion.

@johndgiese I know that Notion can render struck out spaces just fine but the final markdown conversion renders as:
 ~~ STRIKETHROUGH ~~
Instead of something that looks like:
-~~STRIKETHROUGH~~-

With that in mind, I think we should still correct them. What do you think?